build-system/tasks/dist.js

```diff
@@ -276,6 +277,38 @@ function copyParsers() {
   });
 }
 
+/**
+ * Obtain a recursive file listing of a directory
+ * @param {string} dest - Directory to be scanned
+ * @return {Array} - All files found in directory
+ */
+async function walk(dest) {
+  const filelist = [];
+  const files = await fs.readdir(dest);
+
+  for (let i = 0; i < files.length; i++) {
+    const file = `${dest}/${files[i]}`;
+
+    fs.statSync(file).isDirectory()
+      ? Array.prototype.push.apply(filelist, await walk(file))
+      : filelist.push(file);
+  }
+
+  return filelist;
+}
+
+/**
+ * Generate a listing of all files in dist/ and save as dist/files.txt
+ */
+async function generateFileListing() {
+  const distDir = 'dist';
```

Comment:
`gulp dist` writes files to all these directories:
```
build/
dist/
dist.3p/
dist.tools/
```
Would it be useful to generate file listings for these directories too? If so, I suggest calling `walk()` in a loop, and then calling `endBuildStep()` once at the end (see other comment re: logging).